From 0ba01c88fc16a08cde5be1279f59153edcf53cae Mon Sep 17 00:00:00 2001
From: faux123 <reioux@gmail.com>
Date: Sat, 24 Nov 2012 14:20:10 -0800
Subject: [PATCH 048/507] intellidemand: add a run queue persistence to
 determine browsing state

persistence will allow for better browsing state detection rather than just
purely based on run queue depth alone

Signed-off-by: faux123 <reioux@gmail.com>
Signed-off-by: Simarpreet Singh <simar@linux.com>
---
 drivers/cpufreq/cpufreq_intellidemand.c |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/cpufreq/cpufreq_intellidemand.c b/drivers/cpufreq/cpufreq_intellidemand.c
index 8b213e3..8c216b9 100644
--- a/drivers/cpufreq/cpufreq_intellidemand.c
+++ b/drivers/cpufreq/cpufreq_intellidemand.c
@@ -1177,6 +1177,7 @@ static unsigned int calculate_thread_stats (void)
 }
 
 static unsigned int persist_count = 0;
+static unsigned int rq_persist_count = 0;
 
 static void do_dbs_timer(struct work_struct *work)
 {
@@ -1212,9 +1213,18 @@ static void do_dbs_timer(struct work_struct *work)
 		}
 	}
 	if (num_online_cpus() == 2 && rq_info.rq_avg > 38)
+		rq_persist_count++;
+	else
+		if (rq_persist_count > 0)
+			rq_persist_count--;
+
+	if (rq_persist_count > 3) {
 		lmf_browsing_state = false;
+		rq_persist_count = 0;
+	}
 	else
 		lmf_browsing_state = true;
+
 #endif
 
 	//pr_info("Run Queue Average: %u\n", rq_info.rq_avg);
-- 
1.7.9.5

